src/shared/mkfs-util.c

```diff
@@ -487,15 +490,35 @@ int make_filesystem(
                                 return log_oom();
                 }
 
-        } else if (streq(fstype, "vfat"))
+                if (sector_size > 0) {
+                        r = strv_extend(&argv, "-s");
+                        if (r < 0)
+                                return log_oom();
+
```

Comment:
maybe don't bother storing the ret val in `r` if you then propagate log_oom() anyway.

i.e. make this:

```c
if (strv_extend(…) < 0)
        return log_oom();
```

here and everywhere below similar